src/dist/dist.rs

```diff
@@ -926,3 +917,9 @@ fn dl_v1_manifest<'a>(download: DownloadCfg<'a>, toolchain: &ToolchainDesc) -> R
 
     Ok(urls)
 }
+
+fn utc_from_manifest_date(date_str: &str) -> Date<Utc> {
+    let msg = format!("Malformed manifest date: {:?}", date_str);
+    let date = NaiveDate::parse_from_str(date_str, "%Y-%m-%d").expect(&msg);
```

Comment:
Seems odd to do this work rather than the old pattern of:

```rust
foo.unwrap_or_else(||panic!(...))
```

Reply:
I think `unwrap_or_else(||panic!(...))` is the same as `expect(format!(...))` but the expect is more idiomatic. I think I would use or_else to do something else than the normal panic of an unwrap.

Reply:
The thing about the `.expect()` is that the `format!()` is eager, whereas in the `.unwrap_or_else()` the `panic!()` is lazy.  It feels like change for change's sake.